### PR TITLE
Overmap overhaul (step 6)

### PR DESCRIPTION
**Pull request by codemime**
Continues #20003.

This one is a no-brainer.

- Proper flags for overmap terrains (goal of the PR).
- Overmap terrains are now rotatable by default.
- Eliminated discrepancy between rotatable overmap specials and their "fixed" terrains: rotatable overmap specials can't have non-rotatable terrains in them.
- Added a respective check to overmap specials. Linear terrains (roads, sewers, e.t.c.) are considered to be rotatable in this check.
- Fewer lines of JSON.
- Fixed an error in generic factory `bitset` handler (`insert()`/`erase()` were used instead of `set()`/`reset()`).

### CODE REVIEW

File: src/overmap.cpp [hunk 1]
```diff
@@ -674,6 +673,17 @@ void overmap_terrains::check_consistency()
...
+}
+
+void overmap_terrains::load( JsonObject &jo, const std::string &src )
+{
+    terrain_types.load( jo, src );
+}
+
+void overmap_terrains::check_consistency()
+{
+
```
Comment (BrettDong) [inline src/overmap.cpp]: Minor: an empty line at the begining of the function.

**Comment by Coolthulhu** [comment]
It conflicts with all the jsonizations and it seems non-trivially, I'll try the others first.

**Comment by Coolthulhu** [comment]
Needs conflict resolution

**Comment by codemime** [comment]
> Needs conflict resolution

Done.

**Comment by Noctifer-de-Mortem** [comment]
Has the json conflict apply with mods as well? If so, update to the Json API thread would be appreciated.

**Pull request merged**